src/components/Providers/helpers.js

```diff
@@ -6,3 +6,17 @@ export const saveAccountToSettings = ({ client, setting, account }) =>
     account,
     _type: SETTINGS_DOCTYPE
   })
+
+export const sortTimeserieSections = timeserie => {
+  const unorderedSections = timeserie?.aggregation?.sections
```

Comment:
on n'a pas de helper pour récupérer les sections ?

c'est par normal que ce soit unordered par défaut si ? 🤔 C'est bien ordonné dans le geojson non ?

Reply:
> on n'a pas de helper pour récupérer les sections ?

Je ne sais pas, mais je n'y vois guère d'intérêt ici :)

> c'est par normal que ce soit unordered par défaut si ? C'est bien ordonné dans le geojson non ?

je trouve ça également surprenant, car ça semble effectivement bien ordonné en bd, et l'affichage non-ordonné ne semble pas systématique. Mais au moins maintenant on est sûr que le tri est bon

Reply:
du coup j'ai pris le temps de regarder... on n'a pas de helper pour les sections.

A mon avis on devrait plutôt regarder et comprendre pourquoi via des données ordonnées en bd on se retrouve avec du non-ordonné en front, plutôt que de surcharger avec une rustine. C'est quand même un gros indicateur que quelque chose cloche quelque part ailleurs, non ? 

Je valide quand même, si besoin de passer ce fix rapidement... mais on devrait plutôt essayer de corriger le problème initial...

Reply:
D'après fab: 
- "Ok, compris, en fait le bug est intermitant. Selon qu'on ré-ouvre ou pas, l'ordre passe de chaotique à nickel et vis-et-versa" 

y a des fetch policy ? Ça pourrait être lié à cozy-client https://github.com/cozy/cozy-client/issues/790 ? Vous faites des sort quelque part dans vos requêtes ?

Reply:
> Je valide quand même, si besoin de passer ce fix rapidement... mais on devrait plutôt essayer de corriger le problème initial...

Of course, mais c'est bloquant pour la livraison, et je n'ai pas des masses de temps pour explorer un éventuel effet de bord côté app. En plus de ça, faire le tri côté app me paraît de toute façon sain, vu qu'on n'a en réalité aucune garantie que le serveur tracemob nous renverra toujours les données brutes correctement triées. 

> y a des fetch policy ? Ça pourrait être lié à cozy-client https://github.com/cozy/cozy-client/issues/790 ? Vous faites des sort quelque part dans vos requêtes ?

J'y ai pensé, mais ça ne vient pas de là, on ne fait jamais de tri au niveau requête sur les sections (je pense d'ailleurs que ce n'est pas possible avec mango, vu que les sections sont des tableaux d'objets)